ansible/roles/deploy_vpn/tasks/main.yml

```diff
@@ -0,0 +1,87 @@
+---
+
+- name: 'update linux'
+  ansible.builtin.apt:
+    update_cache: yes
+    upgrade: full
+
+- name: 'install ovpn and zip'
+  ansible.builtin.apt:
+    name:
+     - ovpn 
+     - zip
+    state: 'latest'
+
+- name: 'download easy-rsa'
+  ansible.builtin.get_url:
+    url: 'https://github.com/OpenVPN/easy-rsa/archive/master.zip'
+    dest: '/home/vpn_admin'
+
+- name: 'unpack easy-rsa'
+  ansible.builtin.unarchive:
+    src: /home/vpn_admin
+    dest: /home/vpn_admin
+    remote_src: yes
+
+- name: 'create pki and ca, gen DH params'
+  ansible.bultin.command: './easyrsa {{item.command}}'
+  args:
+    chdir: '{{ easyrsa3_path }}'
+    creates: '{{item.creates}}'
+  with_items:
+  - {command: init-pki, creates: '{{ easyrsa3_path }}/pki'}
+  - {command: build-ca nopass, creates: ['{{ easyrsa3_path }}/pki/private/ca.key', '{{ easyrsa3_path }}/pki/ca.crt']}
+  - {command: gen-dh, creates: '{{ easyrsa3_path }}/pki/dh.pem'}
+
+- name: 'gen server req, sign server req'
+  ansible.bultin.command: './easyrsa {{item.command}}'
+  args:
+    chdir: '{{ easyrsa3_path }}'
+    creates: '{{item.creates}}'
+  with_items:
+  - {command: gen-req server nopass, creates: ['{{ easyrsa3_path }}/pki/reqs/server.req','{{ easyrsa3_path }}/pki/private/server.key']}
+  - {command: sign-req server vpn_server, creates: '{{ easyrsa3_path }}/pki/issued/vpn_server.crt'}
+#common name = pet-project-vpn
+
+- name: 'Copy generated/signed and local conf files to etc/openvpn'
+  ansible.builtin.copy:
+    src: '{{item.src}}'
+    dest: '{{item.dest}}'
+    remote_src: yes
+  with_items:
+  - {src: '{{ easyrsa3_path }}/pki/private/ca.key', dest: '/etc/openvpn/ca.key'}
+  - {src: '{{ easyrsa3_path }}/pki/ca.crt', dest: '/etc/openvpn/ca.crt'}
+  - {src: '{{ easyrsa3_path }}/pki/issued/vpn_server.crt', dest: '/etc/openvpn/vpn_server.crt'}
+  - {src: '{{ easyrsa3_path }}/pki/dh.pem', dest: '/etc/openvpn/dh.pem'}
+  - {src: 'openssl.cnf', dest: '/etc/openvpn/openssl.cnf'}
+  - {src: 'server.conf', dest: '/etc/openvpn/server.conf'}
+
+- name: 'copy ca.crt file from server to local'
+  ansible.builtin.fetch:
+    src: '{{ easyrsa3_path }}/pki/ca.crt'
+    dest: '/home/ansible_files_from_server/'
```

Comment:
а если у тебя flat, то тебе не надо сделать `dest: '/home/ansible_files_from_server/ca.crt'`? Он же на то и flat чтобы взять твои dest путь как цель)

Reply:
Дим, я вот это вообще не поняла, если честно.
Документация говорит, что
 ```If dest ends with ‘/’, it will use the basename of the source file, similar to the copy module.
This can be useful if working with a single host, or if retrieving files that are uniquely named per host.
If using multiple hosts with the same filename, the file will be overwritten for each host.```

Раз у меня один хост с уникальным именем, зачем указывать /ca.crt  ?